OrbitCore/LinuxEventTracer.cpp

```diff
@@ -172,13 +193,89 @@ void LinuxEventTracer::Run(bool* a_ExitRequested)
                             CS.m_ThreadId = context_switch.NextTID();
                             CS.m_Time = context_switch.Timestamp();
                             CS.m_ProcessorIndex = context_switch.CPU();
-                            //TODO: Is this correct?
+                            // TODO: Is this correct?
                             CS.m_ProcessorNumber = context_switch.CPU();
                             GCoreApp->ProcessContextSwitch( CS );
                         }
                     }
                     break;
 
+                case PERF_RECORD_FORK:
+                    {
+                        auto fork = ring_buffer.ConsumeRecord<LinuxForkEvent>(header);
+
+                        if (fork.PID() == a_Pid) {
+                            int32_t fd = LinuxPerfUtils::stack_sample_event_open(
+                                    fork.TID(), a_Frequency);
+                            fds.push_back(fd);
+                            ring_buffers.emplace_back(fd);
+                            LinuxPerfUtils::start_capturing(fd);
+                        }
+                    }
+                    break;
+
+                case PERF_RECORD_EXIT:
+                    {
+                        auto exit = ring_buffer.ConsumeRecord<LinuxForkEvent>(header);
+                        if (exit.PID() == a_Pid) {
+                            // TODO: stop event recording, close ring buffer, close file descriptor.
+                        }
+                    }
+                    break;
+
+                case PERF_RECORD_MMAP:
+                    {
+                        // There was a call to mmap with PROT_EXEC, hence
+                        // refresh the maps. This should happen rarely.
+                        ring_buffer.SkipRecord(header);
+                        unwinder.SetMaps(LinuxUtils::ReadMaps(a_Pid));
+                    }
+                    break;
+
+                case PERF_RECORD_SAMPLE:
+                    {
+                        auto sample = ring_buffer.ConsumeRecord<LinuxStackSampleEvent>(header);
+
+                        std::vector<unwindstack::FrameData> frames =
+                            unwinder.Unwind(sample.Registers(), sample.StackDump(), sample.StackSize());
+                        if (frames.empty()) {
+                            continue;
+                        }
+
+                        CallStack cs;
+                        cs.m_ThreadId = sample.TID();
+                        for (const auto& frame : frames) {
```

Comment:
Would it make sense to extract this into a "callstack helper" file? This seems to be a lot of logic inside the case statement.

Reply:
I move it to a separate function for the time being.